src/server/cluster/incoming_slot_migration.h

```diff
@@ -47,12 +47,12 @@ class IncomingSlotMigration {
     return source_id_;
   }
 
-  void ReportError(dfly::GenericError err) {
+  void ReportError(dfly::GenericError err) ABSL_LOCKS_EXCLUDED(error_mu_) {
```

Comment:
//TODO test negative capabilities via `!error_mu_` for stronger invariants. See https://clang.llvm.org/docs/ThreadSafetyAnalysis.html#negative-capabilities

Reply:
It's an experimental flag so I rather not introduce it until it's stable. IMO it's far better though in the static analysis.